test/mocks/stats/mocks.h

```diff
@@ -204,14 +204,19 @@ class MockHistogram : public MockMetric<Histogram> {
   MockHistogram();
   ~MockHistogram() override;
 
-  MOCK_METHOD1(recordValue, void(uint64_t value));
   MOCK_CONST_METHOD0(used, bool());
+  MOCK_CONST_METHOD0(unit, Histogram::Unit());
+  MOCK_METHOD1(recordValue, void(uint64_t value));
+
+  // Histogram
+  std::string unitSymbol() const override { return Histogram::unitSymbol(); }
 
   // RefcountInterface
   void incRefCount() override { refcount_helper_.incRefCount(); }
   bool decRefCount() override { return refcount_helper_.decRefCount(); }
   uint32_t use_count() const override { return refcount_helper_.use_count(); }
 
+  Unit unit_;
```

Comment:
````
unit_{Unit::Unspecified};
```
then you don't have to add those 'unspecified' inits all over the test code.

Reply:
I've checked the test changes and almost all of the instantiations go through store/scope. There's only a handful of the ones using mocks directly and some of them set to `non-Unspecified` anyway, to be consistent I'd rather set them to `Unspecified` explicitly as well. Again it's some usages in `test/extensions/stats_sinks/common/statsd/udp_statsd_test.cc` and `test/server/http/admin_test.cc` and 1 in server test. WDYT?

Reply:
I still think you should default-init the member-variable, either in the ctor or as an initializer.

Reply:
Ah fair point, missed that.